src/Lean/Elab/SyntheticMVars.lean

```diff
@@ -377,12 +385,18 @@ mutual
               kind.logError tacticCode
               reportUnsolvedGoals remainingGoals
             else
-              throwError "unsolved goals\n{goalsToMessageData remainingGoals}")
+              throwError "unsolved goals\n{goalsToMessageData remainingGoals}"
+          if isNoLongerExporting then
+            let mut e ← instantiateExprMVars (.mvar mvarId')
+            if !e.isFVar then
+              e ← mvarId'.withContext do
+                AbstractNestedProofs.visit.mkAuxLemma e |>.run { cache := true } |>.run
```

Comment:
This looks like it’s using “internal” functions from `AbstractNestedProofs`? Why not [Lean.Meta.abstractNestedProofs](https://leanprover-community.github.io/mathlib4_docs/Lean/Meta/AbstractNestedProofs.html#Lean.Meta.abstractNestedProofs)?

Reply:
Thanks for the reminder, I did want to refactor this before I forgot